packages/plugins/persisted-operations/src/index.ts

```diff
@@ -0,0 +1,34 @@
+import { Plugin } from '@envelop/types';
+import { DocumentNode, parse } from 'graphql';
+
+export class NonPersistedOperationError extends Error {}
+
+export interface PersistedOperationsStore {
+  /**
+   * Check if the provided string content can be handled by the store or not.
+   * @param idOrSdl persisted id/hash, or SDL, provided for `parse`
+   */
+  canHandle(idOrSdl: string): boolean;
+  /**
+   * Transfroms a hash/id into a DocumentNode or SDL string. Returns `null` in case of a store miss.
+   * @param id
+   */
+  get(id: string): string | DocumentNode | null;
```

Comment:
This should also allow returning a Promise for async stores 😇

Reply:
Yeah, but it also means that `parse` becomes async, so I wasn't sure if it's the right path to go at the moment, because helix and other pipeline executors will need to adjust to that as well

Reply:
I think maybe store data needs to be fetched ahead of time?

Reply:
TBH I don't have a use-case right now. I could just imagine that if you have a lot lot lot of documents. You don't wanna have all of them in the memory? 

This could still be implemented later, just one thought I had 🤔

Reply:
You don't have to, it can be:

```
class MyStore implements PersistedOperationsStore {
  load() { ... } // async
  canHandle() { ... } // sync
  get() { ... } // sync
}

const store = new MyStore();
await store.load();

const envelopFn = envelop({
  plugins: [usePersistedOperations({ store })]
});
```

Reply:
graphql-ws allows async persisted queries

```ts
useServer(
  {
    onSubscribe: async (_ctx, msg) => {
      const query = await queriesStore.load(msg.payload.query);
      if (!query) {
        // for extra security you only allow the queries from the store
        throw new Error('404: Query Not Found');
      }
      return {
        ...query,
        variableValues: msg.payload.variables, // use the variables from the client
      };
    },
  },
  wsServer,
);
```

Reply:
we can try to add that, it just needs some adjustments to the core of `envelop`, because then the main function is async and the pipeline executors need to adjust to that.